swift/app.py

```diff
@@ -84,3 +95,86 @@ def _receivedMessage(self, channelName: str, msg: str):
             print(f"swift.app._receivedMessage(): {e!r}")
         else:
             self.receivedSlot(channelName, content)
+
+    @pyqtSlot(str, str)
+    def _receivedResult(self, request: str, msg: str):
+        """This is connected to self.swiftcallReturned signal.
+
+        Args:
+            request: The request message that has been sent via 
+              self.swiftcallRequested signal.
+            msg: The received swift-call result message.
+        """
+        try:
+            result = swift.loads(swift.SwiftcallResult, msg)
+        except json.JSONDecodeError as e:
+            print(f"{self}._receivedResult: {e}")
+        else:
+            self.swiftcall.update_result(request, result)
+
+
+class SwiftcallProxy:  # pylint: disable=too-few-public-methods
+    """A proxy for requesting swift-calls conveniently.
+    
+    Every attribute access is proxied, and if you try to call a method of this
+    object, it will emit a swift-call requesting signal instead.
+    If you get an attribute of this object, you will get a callable object which
+    does the same thing as calling a method of this object.
+    """
+
+    def __init__(self, requested: QObject):
+        """
+        Args:
+            requested: A pyqtSignal(str) which will be emitted when a proxied
+              method call is invoked. See BaseApp.swiftcallRequested.
+        """
+        self.requested = requested
+        self.results: Dict[str, swift.SwiftcallResult] = {}
```

Comment:
It looks good to distinguish various reqeusts from their message (`SwiftcallInfo`).
However, I'm worried a little because some requests may be the same.
(In current system, it will NOT occur, but someday..)

What do you think of it? If you're okay, it doesn't matter!

Reply:
I thought about that as well, but I wanted to keep the mechanism simple as possible.
How about tackling that when it becomes an actual problem?

Reply:
I agree with you. I like the simple mechanism now, too!

Reply:
Then, please implement checking if the key already exists.